lib/tasks/plan_validation_reports.rake

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# These rake tasks should be run to generate reports after plans loading.
+# To run all reports please use this rake command: RAILS_ENV=production bundle exec rake plan_validation:reports['2020']
+
+namespace :plan_validation do
+
+  desc "reports generation after plan loading"
+  task :reports, [:plan_year] => :environment do |_task, args|
+    puts "Reports generation started" unless Rails.env.test?
+    puts "Reports generation started for Report1" unless Rails.env.test?
+    Rake::Task['plan_validation:report1'].invoke(args[:plan_year])
+    puts "Reports generation started for Report2" unless Rails.env.test?
+    Rake::Task['plan_validation:report2'].invoke(args[:plan_year])
+  end
+
+  #To run first report: RAILS_ENV=production bundle exec rake plan_validation:report1['2020']
+  desc "Details about Plan Count by Carrier, Coverage and Tier"
+  task :report1, [:plan_year] => :environment do |_task, args|
+    CSV.open("#{Rails.root}/plan_validation_report1_#{args[:plan_year]}.csv", "w", force_quotes: true) do |csv|
+      csv << %w[PlanYearId CarrierId CarrierName PlanTypeCode Tier Count]
+      products = ::BenefitMarkets::Products::Product.by_year(args[:plan_year])
+      products.all.each do |product|
+        plan_year_id = product.active_year
+        carrier_id = product.hios_id[0..4]
+        carrier_name = product.issuer_profile.abbrev
+        plan_type_code = product.kind == :health ? "QHP" : "QDP"
+        tier = product.metal_level_kind
+        product_count = ::BenefitMarkets::Products::Product.by_year(plan_year_id).where(hios_id: /#{carrier_id}/i, metal_level_kind: tier).count
+        csv << [plan_year_id, carrier_id, carrier_name, plan_type_code, tier, product_count]
+      end
+      puts "Successfully generated Plan validation Report1"
+    end
+  end
+
+  #To run second report: RAILS_ENV=production bundle exec rake plan_validation:report2['2020']
+  desc "Rating Area and Age Based Plan Rate Sum"
+  task :report2, [:plan_year] => :environment do |_task, args|
+    plan_year_id = args[:plan_year]
```

Comment:
We need to be able to pass quarterly dates so that we dont update this code every quarter

Reply:
Please use `active_year` instead of `plan_year_id`